### PR TITLE
[WIP] New Bundle directory structure

### PR DESCRIPTION
**Pull request by sadikoff**
It's been a long time since Symfony Bundles were invented, many versions of Symfony changed 
 and finally Symfony 4 and Flex was released. It brought us many changes in App coding style, but what about bundles? Current Bundle structure is out of date. I think it's time to move forward and update it to correspond... lets say flex way =)
### Current:
```
<your-bundle>/
├─ AcmeBlogBundle.php
├─ Controller/
├─ README.md
├─ LICENSE
├─ Resources/
│   ├─ config/
│   ├─ doc/
│   │  └─ index.rst
│   ├─ translations/
│   ├─ views/
│   └─ public/
└─ Tests/
```
### WIP state:
```
<your-bundle>/
├─ docs/
│   └─ index.rst
├─ src/
|   ├─ Controller/
|   ├─ Resources/
|   |   ├─ config/
│   |   ├─ translations/
│   |   ├─ views/
│   |   └─ public/
|   └─ AcmeBlogBundle.php
├─ tests/
├─ README.md
└─ LICENSE
```
### Target

This is what I think it should be in final variant. I'll update this message soon with some info about changes to reach this target.
```
<your-bundle>/
├─ assets/
├─ config/
├─ docs/
│   └─ index.rst
├─ src/
|   ├─ Controller/
|   └─ AcmeBlogBundle.php
├─ templates/
├─ tests/
├─ translations/
├─ README.md
└─ LICENSE
```

### CODE REVIEW

**Comment by yceruto** [comment]
It should close https://github.com/symfony/symfony-docs/issues/5691

File: bundles/best_practices.rst [hunk 1]
```diff
@@ -68,30 +68,31 @@ The basic directory structure of an AcmeBlogBundle must read as follows:
...
 
     <your-bundle>/
-    ├─ AcmeBlogBundle.php
-    ├─ Controller/
+    ├─ src/
+    |   ├─ AcmeBlogBundle.php
+    |   ├─ Controller/
+    |   ├─ Resources/
+    |   |   ├─ config/
+    |   |   ├─ doc/
```
Comment (yceruto) [inline bundles/best_practices.rst]: I think `doc` can be moved to the root and `docs` seems more common https://github.com/symfony/symfony-docs/issues/5691#issuecomment-385971096

**Comment by sadikoff** [comment]
@javiereguiluz Totally agree! This is the goal of this PR, I'm collecting changes to Symfony to get everything work and this PR is first step to change whole bundle structure!

File: bundles/best_practices.rst [hunk 2]
```diff
@@ -68,30 +68,31 @@ The basic directory structure of an AcmeBlogBundle must read as follows:
 .. code-block:: text
 
     <your-bundle>/
-    ├─ AcmeBlogBundle.php
-    ├─ Controller/
+    ├─ src/
+    |   ├─ AcmeBlogBundle.php
```
[the remark this answers is not in the document]
Comment (sadikoff) [reply bundles/best_practices.rst]: It make sense, done!

**Comment by javiereguiluz** [comment]
@sadikoff even if we appreciate your proposal here, I'm afraid I must close it. Symfony Docs only describe what Symfony Code does or recommends ... so we cannot change the recommended bundle dir structure from the docs.

We could create an issue with the label `[RFC]` in https://github.com/symfony/symfony to discuss with the community about these possible changes. I hope you understand. Thanks!

**Pull request closed without merging**